.github/workflows/build-test-publish-images.yml

```diff
@@ -228,10 +250,11 @@ jobs:
         ${{ needs.compute-matrix.outputs.RAPIDS_VER }}\
         ${{ needs.compute-matrix.outputs.ALPHA_TAG }}-\
         cuda${{ matrix.CUDA_VER }}-\
-        py${{ matrix.PYTHON_VER }}"
+        py${{ matrix.PYTHON_VER }}-\
+        ${{ matrix.ARCH }}"
   delete-temp-images:
     if: ${{ !cancelled() && needs.test.result == 'success' }}
-    needs: [compute-matrix, build-multiarch-manifest, test]
+    needs: [compute-matrix, build, test]
```

Comment:
Won't this change mean that for branch builds `delete-temp-images` will not run at all?

Reply:
I don't think so? Because this workflow is invoked on branch builds.

https://github.com/rapidsai/docker/blob/1c27d9245fd9d99ee35981b970acaf10961ca45b/.github/workflows/publish.yml#L3-L6

https://github.com/rapidsai/docker/blob/1c27d9245fd9d99ee35981b970acaf10961ca45b/.github/workflows/publish.yml#L21-L26

 .... BUT it looks like `delete-temp-images` is ALREADY not running on branch builds 🙃 

Look at the most recent one:

https://github.com/rapidsai/docker/actions/runs/10405157692

<img width="1712" alt="Screenshot 2024-08-15 at 12 19 51 PM" src="https://github.com/user-attachments/assets/31248bc4-a68e-4e05-9a9c-af45ae4e3c27">

And that's because it has `[test]` in `needs:` but tests are being skipped, because on branch builds this is run like this:

```text
build_type: branch
run_tests: false
```

Reply:
These requirements can't all be satisfied:

* "do not run `test` on branch builds"
* "only run `delete-temp-images` after the tests pass"
* "always run `delete-temp-images` on branch builds"

Reply:
@raydouglass what do you think about just completely eliminating `delete-temp-images` and instead relying on the other, higher-level cleanup of the `rapidsai/staging` DockerHub repo: https://github.com/rapidsai/workflows/blob/main/.github/workflows/cleanup_staging.yaml

That'd simplify things here a lot, I think.

Reply:
Not necessarily opposed to letting the cleanup happen there, but what is the condition was updated to be "only run delete-temp-images after the test succeeds or is skipped"?

`if: ${{ !cancelled() && (needs.test.result == 'success'  || needs.test.result == 'skipped') }}` maybe?

Reply:
Ok sure, I just pushed a commit with this modified condition.

I do strongly support just dropping `delete-temp-images` entirely. If you'd prefer to discuss that separately from this PR let me know, I can write up an issue.

I'd make CI here work more like it does in other RAPIDS repos, which might reduce the incidence of bugs like those being fixed in this PR.

It'd also remove some complexity... no need to keep https://github.com/rapidsai/docker/blob/branch-24.10/ci/delete-temp-images.sh up to date with changes to the set of image tags or changes in DockerHub's API.

It'd also remove a source of network calls in CI here, which reduces the risk of networking-based job failures.